engine/docs/README.md

```diff
@@ -0,0 +1,23 @@
+# Engine documentation
```

Comment:
Added this README, in order to perhaps organize the documentation a bit in terms of "what-to-read-first"